### ISSUE TITLE
addOnBefore/addOnAfter text not programmatically associated with Input elements (accessibility)

### ISSUE THREAD

**Issue by Watso196** [ant-design/ant-design#46257]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-reproduction-template-forked-vk3yfn)

### Steps to reproduce

- Open the minimal reproduction in Google Chrome
- Open the Devtools inspector (key command F12 or right-click and select Inspect)
- Open the Accessibility menu 
- Use the Selection tool to select either <input> element
- The Accessibility menu's Computed Properties will update to show the <input> has a Name of "", meaning no accessible name has been provided

### What is expected?

The Accessibility menu's Computed Properties should show the Name of the <input> as the value added to the addOnBefore/addOnAfter content, as this content appears to be used as a label for the field

### What is actually happening?

The <input> is not associated with the addOnBefore/addOnAfter content, and therefore the Accessibility menu's Computed Properties show the Name as blank, meaning non-visual users have no context for this field's purpose

| Environment | Info |
| --- | --- |
| antd | 5.12.1 |
| React | 17.0.2 |
| System | MacOS Sonoma 14.1.2 |
| Browser | Chrome 119.0.6045.199 |

---
This bug was encountered while doing accessibility testing on some ANT components in a product we are maintaining
The addOnBefore/addOnAfter elements could be changed to <label> elements, or:
- the addOnBefore/addOnAfter <span> element could be given unique id attribute, then
- the related <input> could be given an aria-labelledby attribute with a value that matches the unique id on the <span>

Either of these methods will provide the contents of the addOnBefore/addOnAfter content as the accessible name of the field

**Comment by afc163** [comment]
There is no `addOnBefore` or `addOnAfter` in your codesandbox.

<img width="707" alt="图片" src="https://github.com/ant-design/ant-design/assets/507615/acbc79fd-6156-422e-a19c-f0aa12743ec6">

**Comment by Watso196** [comment]
@afc163 apologies, I had attached the incorrect link. I've updated the original comment to a codesandbox with `addonBefore` and `addonAfter`

**Comment by prakashk-mallow** [comment]
@afc163 I'm WCAG issue with Form.Item and Input componenet's addOnBefore adding another Form.Item and Select component.

<img width="406" height="83" alt="Image" src="https://github.com/user-attachments/assets/dc4d1987-f1ec-476f-9261-3375685ff7a9" />

WCAG issue details:
Ensure every form element has a label

Fix any of the following:
1. Element does not have an implicit (wrapped) <label>
2. Element does not have an explicit <label>
3. aria-label attribute does not exist or is empty
4. aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
5. Element has no title attribute
6. Element has no placeholder attribute
7. Element's default semantics were not overridden with role="none" or role="presentation"

<img width="1169" height="377" alt="Image" src="https://github.com/user-attachments/assets/b14471af-95e7-4527-9c57-f38f25cf3a37" />

Check and give steps to fix this WCAG issue.
